packages/salesforcedx-vscode-core/src/commands/forceAuthWebLogin.ts

```diff
@@ -29,15 +41,68 @@ class ForceAuthWebLoginExecutor extends SfdxCommandletExecutor<{}> {
   }
 }
 
+export class ForceAuthWebDemoModeLoginExecutor extends SfdxCommandletExecutor<{}> {
```

Comment:
This is the special one that we are using in demo mode. When we use this particular implementation, we will detect if the org that the user has authorized is a prod org and prompt the user appropriately.